src/state/atoms/globalFilterAtom.ts

```diff
@@ -1,38 +1,73 @@
-import { FlagTagsFilter } from '../@types/types';
+import { atom } from 'jotai';
+import { activeModuleAtom } from './activeModuleAtom';
+import { FlagTagsFilter } from '../../@types/types';
 
-export type GlobalFilterWorkloads = {
-  selected?: boolean;
-  page?: number;
-  perPage?: number;
-  isLoaded: boolean;
-  name: 'Workloads';
-  noFilter?: true;
-  tags?: { tag?: CommonTag; count: number }[];
-  items?: any[];
-  count?: number;
-  total?: number;
-};
+export const selectedTagsAtom = atom<FlagTagsFilter>({});
+export const isLoadedAtom = atom<boolean>((get) => {
+  const tags = get(tagsAtom);
+  const sid = get(sidsAtom);
+  const workloads = get(workloadsAtom);
+
+  return tags.isLoaded && sid.isLoaded && workloads.isLoaded;
+});
+export const globalFilterHiddenAtom = atom<boolean>(false);
+export const registeredWithAtom = atom<TagRegisteredWith[number] | undefined>(undefined);
+export const isDisabledAtom = atom<boolean>((get) => get(globalFilterHiddenAtom) || !get(activeModuleAtom));
+export const tagsAtom = atom<GlobalFilterTags>({
+  isLoaded: false,
+  items: [],
+  total: 0,
+  count: 0,
+  page: 1,
+  perPage: 10,
+});
+
+export const sidsAtom = atom<GlobalFilterSIDs>({
```

Comment:
The GlobalFilterSIDs and GlobalFilterTags are the same. I suggest creating a generic type and then creating these specific aliases from it.

The GlobalFilterWorkloads type can then use that as a base and extend it.